test/upsert/failpoint/03-failpoint.td

```diff
@@ -0,0 +1,20 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# This will trigger the failpoint
+$ kafka-ingest format=bytes topic=failpoint key-format=bytes key-terminator=:
+mammal:whale
+
+$ set-regex match="\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d(\.\d\d\d)?" replacement="<TIMESTAMP>"
+
+$ set-from-sql var=source_id
+SELECT id FROM mz_sources WHERE name = 'failpoint'
+
+> select * from mz_internal.mz_source_statuses where id = '${source_id}';
+"${source_id}" failpoint kafka "<TIMESTAMP> UTC" stalled "${arg.error}" <null>
```

Comment:
this can be simplified with a `JOIN`, see https://github.com/MaterializeInc/materialize/blob/main/test/testdrive/source-statistics.td#L80 for an example

(for the record I find writing sql joins absolutely brutal LMAO)

Reply:
lol true. This came from copy-pasta

Reply:
Oh I realized the `source_id` variable is also in the result of the sql. If I do a join, will have to explicitly select columns and not select the source_id in the result. 

The way it's written now might be actually more readable?

Reply:
You don't need the `source_id` in the output, just select the columns you care about, instead of `*`

Reply:
done